src/abstractdataframe/io.jl

```diff
@@ -164,128 +155,143 @@ function html_escape(cell::AbstractString)
+        nrow(df) != 1 &&
+            throw(ArgumentError("rowid may be passed only with a single row data frame"))
+
+        # In this case, if the user does not want to show the row number, then
+        # we must hide the row name column, which is used to display the
+        # `rowid`.
+        if !get(kwargs, :show_row_number, true)
+            row_names = nothing
+            vlines = Int[]
+        else
+            row_names = [string(rowid)]
+            vlines = Int[1]
         end
-        write(io, "</tr>")
+
+        show_row_number = false
     end
-    write(io, "</tbody>")
-    write(io, "</table>")
-    write(io, "</div>")
+
+    pretty_table(io, df;
+                 alignment                 = alignment,
+                 backend                   = Val(:html),
+                 compact_printing          = compact_printing,
+                 formatters                = (_pretty_tables_general_formatter,),
+                 header                    = (names_str, types_str),
+                 header_alignment          = :l,
+                 header_cell_titles        = (nothing, types_str_complete),
+                 highlighters              = (_PRETTY_TABLES_HTML_HIGHLIGHTER,),
+                 max_num_of_columns        = mxcol,
+                 max_num_of_rows           = mxrow,
+                 minify                    = true,
+                 nosubheader               = !eltypes,
+                 row_name_column_title     = "Row",
+                 row_names                 = row_names,
+                 row_number_alignment      = :r,
+                 row_number_column_title   = "Row",
+                 show_omitted_cell_summary = true,
+                 show_row_number           = show_row_number,
+                 standalone                = false,
+                 table_class               = "data-frame",
+                 table_div_class           = "data-frame",
+                 top_left_str              = title,
+                 top_right_str_decoration  = HtmlDecoration(font_style = "italic"),
+                 vcrop_mode                = :middle,
+                 wrap_table_in_div         = true,
+                 kwargs...)
+
+    return nothing
 end
 
-function Base.show(io::IO, mime::MIME"text/html", dfr::DataFrameRow;
-                   summary::Bool=true, eltypes::Bool=true)
+function Base.show(io::IO, mime::MIME"text/html", dfr::DataFrameRow; kwargs...)
     r, c = parentindices(dfr)
-    summary && write(io, "<p>DataFrameRow ($(length(dfr)) columns)</p>")
-    _show(io, mime, view(parent(dfr), [r], c), summary=false, eltypes=eltypes, rowid=r)
+    title = "DataFrameRow ($(length(dfr)) columns)"
+    _show(io, mime, view(parent(dfr), [r], c); rowid=r, title=title, kwargs...)
 end
 
-function Base.show(io::IO, mime::MIME"text/html", dfrs::DataFrameRows;
-                   summary::Bool=true, eltypes::Bool=true)
+function Base.show(io::IO, mime::MIME"text/html", dfrs::DataFrameRows; kwargs...)
```

Comment:
same here (I will not repeat it, but checking `kwargs` probably applies everywhere)